pydantic/fields.py

```diff
@@ -181,6 +184,9 @@ def _populate_sub_fields(self):
         elif issubclass(origin, Set):
             self.type_ = self.type_.__args__[0]
             self.shape = Shape.SET
+        elif issubclass(origin, Json):
```

Comment:
I don't think you need to many any changes in `fields.py`

Reply:
The reason behind changes in `fields.py` is that `Json[List]` had `__origin__` attribute, thus in `_populate_sub_fields` I was getting AssertionError after `origin` was compared against List, Dict etc.